src/main.c

```diff
@@ -112,7 +114,7 @@ static int read_exact (int fd, void *buffer, size_t buffer_size) /* {{{ */
 		status = read (fd, buffer_ptr, want_bytes);
 		if (status < 0)
 		{
-			if (errno == EAGAIN || errno == EINTR)
+			if (errno == EINTR)
 				continue;
 			else
 				return (status);
```

Comment:
Please consider returning `errno` instead, so that the caller has access to the reason for failure.